.github/workflows/publish.yml

```
@@ -9,31 +9,31 @@ on:
         default: true
         type: boolean
 
+permissions:
+  contents: read
+  id-token: write
+
 jobs:
   publish:
     runs-on: ubuntu-latest
-    permissions:
-      contents: read
-      id-token: write
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v6
         with:
           node-version: 22
           registry-url: https://registry.npmjs.org
+
+      - name: Update npm for OIDC
+        run: npm install -g npm@latest
       - run: npm ci
 
```

Since Node 24 just became the [Active LTS release](https://github.com/nodejs/Release), I propose moving to Node 24, which has a compatible npm version, and then we can skip the npm install.

```suggestion
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v6
        with:
          node-version: 24
          registry-url: https://registry.npmjs.org

      - run: npm ci

```